lib/modules/asset/AssetService.ts

```diff
@@ -598,6 +600,78 @@ export class AssetService extends DigitalTwinService {
     return replacedAssets;
   }
 
+  /**
+   * Retrieve locales with the specified model and search all assets related to the model to update the locales.
+   * The operation is historized.
+   * @param request
+   * @param engineGroup
+   * @param model
+   */
+  public async updateModelLocales(
+    request: KuzzleRequest,
+    engineGroup: string,
+    model: string,
+  ) {
+    const { result: resultModel } = await this.sdk.query({
+      action: "getAsset",
+      body: {},
+      controller: "device-manager/models",
+      engineGroup,
+      model,
+    });
+
+    const locales = resultModel._source.asset.locales;
+
+    const engines = await ask<AskEngineList>("ask:device-manager:engine:list", {
+      group: engineGroup,
+    });
+
+    const res = { errors: [], successes: [] };
+
+    for (const engine of engines) {
+      const resUpdateByQuery =
+        await this.sdk.document.updateByQuery<AssetContent>(
+          engine.index,
+          "assets",
+          {
+            bool: {
+              must: [
+                {
+                  term: {
+                    model: "Container",
+                  },
+                },
+              ],
+            },
+          },
+          {
+            modelLocales: locales,
+          },
+        );
+
+      res.successes.push(...resUpdateByQuery.successes);
+      res.errors.push(...resUpdateByQuery.errors);
+
```

Comment:
you should probably add request.status = res.status to maque sure you return the same status code as updateByQuery which may résult in 200 alldoc updated, 206, some docs where updated, 4XX error, no docs updated!